assets/js/aplus.js

```diff
@@ -666,27 +670,18 @@ $(function() {
  * Note! Translation requires that at least 'hl' query parameter is copied
  */
 (function(window) {
-    window.addEventListener('click', function(event) {
-        if (
-            event.target.nodeName === 'A'
-            && event.target.href
-            && event.target.protocol === window.location.protocol
-            && event.target.host === window.location.host // hostname:port
-            && (event.target.search === '' || event.target.search === '?') // no search
-        ) {
-            event.preventDefault(); // stop default redirect
-            event.target.search = window.location.search; // copy query params
-            if (event.target.getAttribute('target') === '_blank') {
-                // Open a new window or tab.
-                window.open(
-                    event.target.href,
-                    '_blank',
-                    'location,menubar,resizable,scrollbars,status,noopener');
-            } else {
-                window.location.href = event.target.href; // redirect the current window
-            }
-        }
-    }, false);
+  const addQueryParameter = (event) => {
+  if (
+      event.target.protocol === window.location.protocol
+      && event.target.host === window.location.host // hostname:port
+      && window.location.search.includes('?hl=en') // hl search
+  ) {
+      event.target.search = window.location.search; // copy query params
+  }
+}
+
+$(document).on('click', 'a[href]', addQueryParameter);
```

Comment:
I would first assume that adding the event handler like this does not work for any link that is added after the initial page load by JS. For example, the exercise descriptions in chapters are loaded with AJAX. They are not part of the initial page load. The original event handler in the whole window or body element works even if the DOM is modified later on.

Reply:
Actually, the event handler is still attached to the document and it should work even after new `<a>` elements are added to the DOM. I somehow misread this. Is there any difference between the **window** and the **document** in event handlers?

Reply:
It shouldn't be any difference between **window** and **document**. **document** is just a child of the window object